regex-role-changer/main.py

```diff
@@ -0,0 +1,185 @@
+        for project_name in matching_projects:
+            # Пробуем найти уже существующие назначения на роли. Если они есть, то модифицируем их
+            modified_user_projects = []
+            created_user_projects = []
+            for idx, user_project in enumerate(users):
+                if user_project['name'] != username:
+                    continue
+                try:
+                    project_index = [proj['projectName'] for proj in user_project['projectMemberInfos']].index(project_name)
+                    modified_user_project = users[idx]['projectMemberInfos']
+                    # Даже если проектные роли совпадают, их нужно добавить в запрос, иначе они будут удалены
+                    # Но для отслеживания именно изменённых проектов кладём их в отдельный список
+                    if modified_user_project[project_index]['roleId'] != role_id:
+                        changed_project_ids.append(modified_user_project[project_index]['id'])
+                        changed_user_names.append(username)
+                    modified_user_project[project_index]['roleId'] = role_id
+                    user_modified_projects[username] = modified_user_project
+                except ValueError:
+                    # У пользователя нет этого проекта в ролях, редактировать нечего, поэтому создаём новый экземпляр
+                    project_list_to_create = new_roles.get(username, [])
+                    new_project = {
+                        'id': increment_and_return_id(),
+                        'roleId': role_id,
+                        'projectId': first(project['id'] for project in projects_raw if project['name'] == project_name),
+                        'userId': first(user['userId'] for user in users if user['name'] == username),
+                        'projectName': project_name # Временное поле для удобства на этапе вывода
+                    }
+                    project_list_to_create.append(new_project)
+                    new_roles[username] = project_list_to_create
+
+# Вывод новых проектных ролей для пользователей
+print("====== Новые проектные роли для пользователей ======")
+for username, projects in new_roles.items():
+    print(f"## {username}")
+    for project in projects:
+        print(f"{project['projectName']}: {roleid_to_name(project['roleId'])}")
+        del project['projectName']
+    print('\n')
+
+# Вывод изменённых проектных ролей для пользователей
+print("====== Изменённые проектные роли для пользователей ======")
+for username, projects in user_modified_projects.items():
+    if not username in changed_user_names:
+        continue
+    print(f"## {username}\n")
+    for project in projects:
+        if project['id'] in changed_project_ids:
+            print(f"{project['projectName']}: {roleid_to_name(project['roleId'])}")
+
+print("Изменить? [y/N]")
+
+if input().lower() not in ['y', 'yes', 'д', 'да']:
+    sys.exit(0)
```

Comment:
Я бы тут сделал вывод сообщения о том, что операция прервана. А то может показаться, что изменения применились. Мало ли кто-то нечаянно введет у (русская буква у)

```suggestion
    print("Отмена операции, изменения не будут применены")
    sys.exit(0)
```